src/batchUploader.ts

```diff
@@ -172,29 +200,52 @@ export class BatchUploader {
     private async prepareAndUpload(triggerFuture: boolean, useBeacon: boolean) {
         const currentUser = this.mpInstance.Identity.getCurrentUser();
 
-        const currentEvents = this.pendingEvents;
+        // const currentEvents = this.pendingEvents;
+        // this.pendingEvents = [];
+
+        // TODO: Retrieve and Purge events from Event Vault
+        const currentEvents = this.eventVault.retrieveItems();
+        // Should this really purge, or just empty local storage?
```

Comment:
Is the difference between purging and emptying local storage that it will be an array or empty? if so, i think it should just be purged into an empty array.